modules/tm/t_reply.c

```diff
@@ -443,6 +444,21 @@ static int _reply_light( struct cell *trans, char* buf, unsigned int len,
 	if(trans->uas.request && trans->uas.request->flags&tcp_no_new_conn_rplflag)
 		tcp_no_new_conn = 1;
 
+	if(ref_script_route_is_valid(tm_local_reply)) {
+		LM_DBG("Found Local-Reply Route...\n");
+		LM_DBG("Message:\n-----------------\n%.*s\n--------------------\n", len, buf);
+		memset(&dummy_msg, 0, sizeof(struct sip_msg));
+		dummy_msg.buf = buf;
+		dummy_msg.len = len;
+
+		if (parse_msg(buf, len, &dummy_msg) == 0) {
+			LM_DBG("Parsed Message, executing Local-Reply Route with Message...\n");
+			run_top_route(sroutes->onreply[tm_local_reply->idx], &dummy_msg);
+		}
+		free_sip_msg(&dummy_msg);
+	}
```

Comment:
ok, this is something new, to have a route run when opensips generates (not proxying). Still, will this be triggered for any locally generated reply, like even for the 100 Trying?
Secondly, more confirming actually, this route is NOT intended for changing anything on the reply, but only for inspection purposes, right ?
And...my I ask for the typical example of using such reply route? :)